### DISCUSSION TITLE
Building on HyperChains

### DISCUSSION THREAD

**Discussion by castig** [hirosystems/stacks-hyperchains#55] [General]
Hi team! I just watched Aaron's [YouTube talk](https://www.youtube.com/watch?v=GXOuRA011MA), and read the [blog post](https://www.hiro.so/blog/introducing-hyperchains-by-hiro) on Hyperchains. Amazing work! Super exciting stuff. 

I see the MVP is slated for June 2022, and have two questions: 

1) Launch — I'm currently building on STX, and was curious if the Hyperchain will in fact be ready for prime time come June/July. Could I confidently deploy a Clarity contract on Hyperchain, or will it be in "beta/testing" for a while? 

2) TPS — What is the current transactions per second (tps) on Stacks, and where will hyperchains bring us? I feel like the tps metric is often cited as a huge competitive advantage by other L2s, so I'd be curious where hyperchains will fall in comparison. 

Thanks so much!

**Comment by diwakergupta** [comment]
FYI @kantai @saralab

**Comment by kantai** [comment]
1. **Launch** -- The current timeline for hyperchains development is to launch a hosted deployment of hyperchains on the Stacks testnet in May. This version will be single-L2 miner, with support for hyperchain deposits. The testnet deployment will be iterated on to include support for withdrawals, improvements to latency, and throughput. An upgrade to support federated hyperchains will come after this, as well as any work to improve the implementation of hyperchains learned during the deployment to testnet. A deployment to mainnet could be done at any time -- developers are free to run a hyperchain on mainnet using the open-source code in this repository, and as we learn about best practices for deployment in testnet, we'll certainly work to upstream scripts and helpful tooling to this repository. This should make deployment rather easy. When it comes to making hyperchains ready for prime time, that is a question that depends ultimately on the use case. For deploying a more app-specific hyperchain that is intended for a low-risk application, a developer could be happy with running a single-miner hyperchain. For a general-purpose hyperchain where the intended use case is higher risk, then developers would probably want to wait until they could run a federated hyperchain.
2.  **TPS** -- The throughput of a hyperchain is entirely configurable by the operator of that hyperchain. In the first implementation of hyperchains on testnet, we'll use a very conservative block limit, which would provide around 4x the throughput of the current Stacks network on the hyperchain. As the implementation matures, we can expect higher throughput. As for _latency_, transactions should confirm much more confidently in hyperchain "microblocks", than the corresponding microblocks of L1. The first deployment of a hyperchain to testnet will not make much use of this, however, as the implementation progresses, we expect latencies of around 30 seconds.

**Reply by biubiuboy-b** [reply]
Hello developers, take the liberty to ask a question, why is the Hyperchain testnet not using the maximum theoretical throughput? Why are testnets more conservative than post-launch?
